service/k8sclient/internal/kube/kube.go

```diff
@@ -0,0 +1,280 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	defaultApiUriPath  = "/api"
+	defaultApisUriPath = "/apis"
+)
+
+type configGetter struct {
+	kubeconfig string
+}
+
+type Client struct {
+	clientset  *kubernetes.Clientset
+	restConfig *rest.Config
+}
+
+func NewConfigGetter(kubeconfig string) *configGetter {
+	return &configGetter{kubeconfig: kubeconfig}
+}
+
+// LoadFromString takes a kubeconfig and deserializes the contents into Config object
+func (g *configGetter) loadFromString() (*clientcmdapi.Config, error) {
+	config, err := clientcmd.Load([]byte(g.kubeconfig))
+	if err != nil {
+		return nil, err
+	}
+
+	// set LocationOfOrigin on every Cluster, User, and Context
+	for key, obj := range config.AuthInfos {
+		config.AuthInfos[key] = obj
+	}
+	for key, obj := range config.Clusters {
+		config.Clusters[key] = obj
+	}
+	for key, obj := range config.Contexts {
+		config.Contexts[key] = obj
+	}
+
+	if config.AuthInfos == nil {
+		config.AuthInfos = map[string]*clientcmdapi.AuthInfo{}
+	}
+	if config.Clusters == nil {
+		config.Clusters = map[string]*clientcmdapi.Cluster{}
+	}
+	if config.Contexts == nil {
+		config.Contexts = map[string]*clientcmdapi.Context{}
+	}
+
+	return config, nil
+}
+
+func NewFromIncluster() (*Client, error) {
+	c, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(c)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{clientset: clientset, restConfig: c}, nil
+}
+
+func NewFromKubeConfig(kubeConfigPath string) (*Client, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{clientset: clientset, restConfig: config}, nil
+}
+func NewFromKubeConfigObject(kubeconfig string) (*Client, error) {
+	config, err := clientcmd.BuildConfigFromKubeconfigGetter("", NewConfigGetter(kubeconfig).loadFromString)
+	if err != nil {
+		return nil, err
+	}
+	config.QPS = 350
+	config.Burst = 500
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{clientset: clientset, restConfig: config}, nil
+}
+
+func (c *Client) resourceClient(gv schema.GroupVersion) (rest.Interface, error) {
+	cfg := c.restConfig
+	cfg.ContentConfig = resource.UnstructuredPlusDefaultContentConfig()
+	cfg.GroupVersion = &gv
+	if len(gv.Group) == 0 {
+		cfg.APIPath = defaultApiUriPath
+	} else {
+		cfg.APIPath = defaultApisUriPath
+	}
+	return rest.RESTClientFor(cfg)
+}
+func (c *Client) Delete(ctx context.Context, rawObj interface{}) error {
```

Comment:
I think that I need to change rawObj interface{} to runtime.Object, but I'm not sure about it